rialto_airflow/snapshot.py

```diff
@@ -11,16 +13,81 @@ class Snapshot:
     """
 
     path: Path
-    time: str
+    timestamp: str
     database_name: str
-    authors_csv: Path
 
-    def __init__(self, path, database_name=None):
-        now = datetime.datetime.now()
-        self.time = now.strftime("%Y%m%d%H%M%S")
+    @classmethod
+    def create(cls, data_dir: Path, database_name: str | None = None):
+        """
+        Create a Snapshot in the given root data directory. Optionally you can
+        override the database name, which is really only useful in testing.
+        """
+        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
 
-        self.path = Path(path) / "snapshots" / self.time
-        self.path.mkdir(parents=True)
+        path = Path(data_dir) / "snapshots" / timestamp
+        path.mkdir(parents=True, exist_ok=True)
 
-        self.database_name = database_name or f"rialto_{self.time}"
-        self.authors_csv = self.path / "authors.csv"
+        database_name = database_name or f"rialto_{timestamp}"
+
+        return cls(path, timestamp, database_name)
+
+    @classmethod
+    def get_latest(cls, data_dir: Path):
+        """
+        Returns the most recent completed Snapshot.
+        """
+        data_dir = Path(data_dir)
+
+        snapshots_dir = data_dir / "snapshots"
+        if not snapshots_dir.is_dir():
+            return None
+
+        snapshot_names = [p.name for p in snapshots_dir.iterdir()]
+
+        for snapshot_name in reversed(sorted(snapshot_names)):
+            # ignore things that might be here that aren't the right format
+            if not re.match(r"^\d{14}$", snapshot_name):
+                continue
+
+            # if the snapshot.json file is there it is complete
+            snapshot_path = snapshots_dir / snapshot_name
+            if not (snapshot_path / "snapshot.json").is_file():
+                continue
+
+            # TODO: store the database name in snapshot.json?
+            database_name = f"rialto_{snapshot_path.name}"
+
+            return Snapshot(snapshot_path, snapshot_name, database_name)
+
+        # no complete snapshot was found
+        return None
+
```

Comment:
This method marks a Snapshot as complete. It is called at the end of the harvest DAG.